crates/biome_cli/src/reporter/mod.rs

```
@@ -0,0 +1,206 @@
+        self.duration = value;
+        self
+    }
+    pub fn with_errors(mut self, value: u32) -> Self {
+        self.errors = value;
+        self
+    }
+
+    pub fn with_warnings(mut self, value: u32) -> Self {
+        self.warnings = value;
+        self
+    }
+
+    pub fn with_skipped(mut self, value: usize) -> Self {
+        self.skipped = value;
+        self
+    }
+
+    pub fn with_suggested_fixes_skipped(mut self, value: u32) -> Self {
+        self.suggested_fixes_skipped = value;
+        self
+    }
+
+    pub fn with_diagnostics_not_printed(mut self, value: u32) -> Self {
+        self.diagnostics_not_printed = value;
+        self
+    }
+
+    pub fn changed(&self) -> usize {
+        self.changed
+    }
+    pub fn unchanged(&self) -> usize {
+        self.unchanged
+    }
+    pub fn duration(&self) -> Duration {
+        self.duration
+    }
+    pub fn errors(&self) -> u32 {
+        self.errors
+    }
+    pub fn warnings(&self) -> u32 {
+        self.warnings
+    }
+    pub fn skipped(&self) -> usize {
+        self.skipped
+    }
+    pub fn suggested_fixes_skipped(&self) -> u32 {
+        self.suggested_fixes_skipped
+    }
+    pub fn diagnostics_not_printed(&self) -> u32 {
+        self.diagnostics_not_printed
+    }
+}
+
+/// When using this trait, the type that implements this trait is the one that holds the read-only information to pass around
+pub trait Reporter {
+    /// Writes the summary using the underling visitor
+    fn write(&mut self, visitor: &mut dyn ReporterVisitor) -> io::Result<()>;
+}
+
+/// When using this trait, the type that implements this trait is the one that will **write** the data, ideally inside a buffer
+pub trait ReporterVisitor {
+    /// Writes the summary in the underling writer
+    fn report_summary(
+        &mut self,
+        traversal_mode: &Execution,
+        summary: &TraversalSummary,
+    ) -> io::Result<()> {
+        let _ = (summary, traversal_mode);
+        Ok(())
```

I'm a little hesitant whether silently swallowing everything is a good default behavior for a reporter. I would lean towards leaving the methods abstract and letting reporters choose more explicitly what they want to report vs. what they want to ignore.